File: src/app/customer.ts
export class Customer {
    id: number;
    firstName: string;
    lastName: string;
    username: string;
    password: string;
    emailAddress: string;

    constructor(id:number, firstName: string, lastName: string, username: string, password: string, emailAddress: string){
        this.id = id;
        this.firstName = firstName;
        this.lastName = lastName;
        this.username = username;
        this.password = password;
        this.emailAddress = emailAddress;
    }
}

File: shop/src/main/java/proiect/sd/shop/entity/Customer.java
package proiect.sd.shop.entity;


import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.List;

@Data
@Entity
@NoArgsConstructor
@AllArgsConstructor
public class Customer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer customerId;
    private String firstName;
    private String lastName;
    private String username;
    private String password;
    private String emailAddress;

    @JsonIgnore
    @OneToMany(mappedBy = "customer")
    private List<Orders> orders;

    public Customer(String firstName, String lastName, String username, String password, String emailAddress){
        this.firstName = firstName;
        this.lastName = lastName;
        this.username = username;
        this.password = password;
        this.emailAddress = emailAddress;
    }
}